In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv("data/customer_churn_dataset-training-master.csv")
df.head()

,CustomerID,Age,Gender,Tenure,Usage Frequency,Support Calls,Payment Delay,Subscription Type,Contract Length,Total Spend,Last Interaction,Churn
0,2.0,30.0,Female,39.0,14.0,5.0,18.0,Standard,Annual,932.0,17.0,1.0
1,3.0,65.0,Female,49.0,1.0,10.0,8.0,Basic,Monthly,557.0,6.0,1.0
2,4.0,55.0,Female,14.0,4.0,6.0,18.0,Basic,Quarterly,185.0,3.0,1.0
3,5.0,58.0,Male,38.0,21.0,7.0,7.0,Standard,Monthly,396.0,29.0,1.0
4,6.0,23.0,Male,32.0,20.0,5.0,8.0,Basic,Monthly,617.0,20.0,1.0


### 
The dataset was loaded successfully and contains customer demographic,subscription, usage behaviour and spending information together with the target variable Churn.

In [3]:
df.shape

(440833, 12)

###
The dataset contains 440833 rows and 12 columns.
Each row represents an individual customer.

In [5]:
df.columns.tolist()

['CustomerID',
 'Age',
 'Gender',
 'Tenure',
 'Usage Frequency',
 'Support Calls',
 'Payment Delay',
 'Subscription Type',
 'Contract Length',
 'Total Spend',
 'Last Interaction',
 'Churn']

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 440833 entries, 0 to 440832
Data columns (total 12 columns):
 #   Column             Non-Null Count   Dtype  
---  ------             --------------   -----  
 0   CustomerID         440832 non-null  float64
 1   Age                440832 non-null  float64
 2   Gender             440832 non-null  object 
 3   Tenure             440832 non-null  float64
 4   Usage Frequency    440832 non-null  float64
 5   Support Calls      440832 non-null  float64
 6   Payment Delay      440832 non-null  float64
 7   Subscription Type  440832 non-null  object 
 8   Contract Length    440832 non-null  object 
 9   Total Spend        440832 non-null  float64
 10  Last Interaction   440832 non-null  float64
 11  Churn              440832 non-null  float64
dtypes: float64(9), object(3)
memory usage: 40.4+ MB


###
The dataset contains 440,833 observations and 12 variables.

There are 9 numerical variables and 3 categorical variables.

The categorical variables are:
- Gender
- Subscription Type
- Contract Length

Most variables have 440,832 non-null values, indicating that the dataset
contains one missing value in each column.

In [7]:
df.describe()

,CustomerID,Age,Tenure,Usage Frequency,Support Calls,Payment Delay,Total Spend,Last Interaction,Churn
count,440832.000000,440832.000000,440832.000000,440832.000000,440832.000000,440832.000000,440832.000000,440832.000000,440832.000000
mean,225398.667955,39.373153,31.256336,15.807494,3.604437,12.965722,631.616223,14.480868,0.567107
std,129531.918550,12.442369,17.255727,8.586242,3.070218,8.258063,240.803001,8.596208,0.495477
min,2.000000,18.000000,1.000000,1.000000,0.000000,0.000000,100.000000,1.000000,0.000000
25%,113621.750000,29.000000,16.000000,9.000000,1.000000,6.000000,480.000000,7.000000,0.000000
50%,226125.500000,39.000000,32.000000,16.000000,3.000000,12.000000,661.000000,14.000000,1.000000
75%,337739.250000,48.000000,46.000000,23.000000,6.000000,19.000000,830.000000,22.000000,1.000000
max,449999.000000,65.000000,60.000000,30.000000,10.000000,30.000000,1000.000000,30.000000,1.000000


###
The customers in the dataset have an average age of approximately 39 years, with ages ranging from 18 to 65 years.
Customers have an average tenure of 31 months, with some customers being new customers (1 month) and others remaining for up to 60 months.
The average usage frequency is approximately 16 interactions, suggesting moderate customer engagement.
Customers make an average of 3.6 support calls, although some customers require as many as 10 support interactions.
The average payment delay is approximately 13 days, with delays ranging from 0 to 30 days.
Customers spend an average of 631 monetary units, with total spending ranging from 100 to 1000.
The average value of the target variable Churn is 0.567, indicating that approximately 56.7% of customers in the dataset have churned.

### target variable (churn) distribution

In [9]:
df["Churn"].value_counts()
df["Churn"].value_counts(normalize=True)*100

Churn
1.0    56.71072
0.0    43.28928
Name: proportion, dtype: float64

###
The dataset contains 249,999 churned customers and 190,833 retained customers.

This indicates that the dataset contains examples from both classes,
making it suitable for supervised binary classification modelling.

Approximately 56.7% of customers in the dataset have churned, while
43.3% have remained with the company.

Although the class distribution is not perfectly balanced, the imbalance
is relatively moderate compared to problems such as credit card fraud
detection. Techniques such as SMOTE may still be explored to determine
whether they improve model performance, particularly recall for the
minority class.
